src/incflo_advance.cpp

```diff
@@ -24,32 +25,24 @@ void incflo::Advance()
         m_t_new[lev] = m_time.new_time();
     }
 
-    copy_from_new_to_old_velocity();
-    copy_from_new_to_old_density();
-    copy_from_new_to_old_tracer();
+    m_repo.get_field("velocity").advance_states();
+    m_repo.get_field("density").advance_states();
+    m_repo.get_field("tracer").advance_states();
 
-    int ng = nghost_state();
-    for (int lev = 0; lev <= finest_level; ++lev) {
-        fillpatch_velocity(lev, m_t_old[lev], m_leveldata[lev]->velocity_o, ng);
-        fillpatch_density(lev, m_t_old[lev], m_leveldata[lev]->density_o, ng);
-        if (m_advect_tracer) {
-            fillpatch_tracer(lev, m_t_old[lev], m_leveldata[lev]->tracer_o, ng);
-        }
-    }
+    m_repo.get_field("velocity",amr_wind::FieldState::Old).fillpatch(m_time.current_time());
+    m_repo.get_field("density",amr_wind::FieldState::Old).fillpatch(m_time.current_time());
+    m_repo.get_field("tracer",amr_wind::FieldState::Old).fillpatch(m_time.current_time());
 
-   for (auto& pp: m_physics)
-       pp->pre_advance_work();
+    for (auto& pp: m_physics)
+        pp->pre_advance_work();
     
     ApplyPredictor();
 
     if (!m_use_godunov) {
-        for (int lev = 0; lev <= finest_level; ++lev) {
-            fillpatch_velocity(lev, m_t_new[lev], m_leveldata[lev]->velocity, ng);
-            fillpatch_density(lev, m_t_new[lev], m_leveldata[lev]->density, ng);
-            if (m_advect_tracer) {
-                fillpatch_tracer(lev, m_t_new[lev], m_leveldata[lev]->tracer, ng);
-            }
-        }
+
+        m_repo.get_field("velocity",amr_wind::FieldState::New).fillpatch(m_time.current_time());
+        m_repo.get_field("density",amr_wind::FieldState::New).fillpatch(m_time.current_time());
+        m_repo.get_field("tracer",amr_wind::FieldState::New).fillpatch(m_time.current_time());
```

Comment:
To be fully consistent with the old code, we should use `m_time.new_time()` instead of `m_time.current_time()` as the old code is using `m_t_new` and not `m_t_old`

Reply:
good catch, if we are moving to the version of fill patch that does not interpolate in time, should we just remove time as an input?

Reply:
The AMReX routines still require a time input even if we are passing just a single field. We can provide `Field::fillpatch` overload that takes no `time` argument and then use `m_time.current_time()` for AMReX routines, current not sure if that is the best approach.